File: 0242-valid-anagram/solution.c
bool isAnagram(char* s, char* t) {
    unsigned int bitStr1[26] = {0};
    for (; *s != '\0'; s++) {
        bitStr1[*s - 'a']++;
    }
    for (; *t != '\0'; t++) {
        bitStr1[*t - 'a']--;
        if (bitStr1[*t - 'a']  < 0)
            return false;
    }
    for (int i = 0; i < 26; i++) {
        if (bitStr1[i] != 0) return false;
    }
    return true;
}

File: 0242-valid-anagram/solution.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if (len(s) != len(t)):
            return False
        counts = {}
        for c in s:
            counts[c] = counts.get(c, 0) + 1
        for c in t:
            counts[c] = counts.get(c, 0) - 1
            if counts[c] < 0:
                return False
        return all(count == 0 for count in counts.values())
